File: jobs/2026-04-23__11-07-16/b2b_org_settings__ijeeDTv/artifacts/update_org.js
'use strict';

const stytch = require('stytch');

const projectId = process.env.STYTCH_PROJECT_ID;
const secret = process.env.STYTCH_SECRET;

if (!projectId || !secret) {
  console.error('Error: STYTCH_PROJECT_ID and STYTCH_SECRET environment variables must be set.');
  process.exit(1);
}

const organizationId = process.argv[2];

if (!organizationId) {
  console.error('Error: organization_id must be provided as the first command-line argument.');
  console.error('Usage: node update_org.js <organization_id>');
  process.exit(1);
}

const client = new stytch.B2BClient({
  project_id: projectId,
  secret: secret,
});

async function updateOrganization() {
  const response = await client.organizations.update({
    organization_id: organizationId,
    auth_methods: 'RESTRICTED',
    allowed_auth_methods: ['sso', 'magic_link'],
    email_jit_provisioning: 'RESTRICTED',
  });

  console.log('Organization updated successfully');
  return response;
}

updateOrganization().catch((err) => {
  console.error('Failed to update organization:', err);
  process.exit(1);
});

File: jobs/2026-04-24__19-56-51/b2b_update_organization__6iXvTE4/artifacts/update_org.py
import os
import stytch

# Read credentials from environment variables
project_id = os.environ["STYTCH_B2B_PROJECT_ID"]
secret = os.environ["STYTCH_B2B_SECRET"]

# Read trial_id from file
with open("/logs/trial_id") as f:
    trial_id = f.read().strip()

org_name = f"test-org-{trial_id}"
org_slug = f"test-org-{trial_id}"

# Initialize the Stytch B2B client
client = stytch.B2BClient(project_id=project_id, secret=secret)

# Step 1: Create the organization
print(f"Creating organization: {org_name}")
create_resp = client.organizations.create(
    organization_name=org_name,
    organization_slug=org_slug,
)
print(f"Create response: {create_resp}")

organization_id = create_resp.organization.organization_id
print(f"Created organization ID: {organization_id}")

# Step 2: Update the organization to restrict auth methods
print("Updating organization auth methods to RESTRICTED (sso, magic_link)...")
update_resp = client.organizations.update(
    organization_id=organization_id,
    auth_methods="RESTRICTED",
    allowed_auth_methods=["sso", "magic_link"],
)
print(f"Update response: {update_resp}")

print("Done.")
